src/contracts/GPv2Settlement.sol

```diff
@@ -100,24 +106,128 @@ contract GPv2Settlement {
     /// @param clearingPrices An array of clearing prices where the `i`-th price
     /// is for the `i`-th token in the [`tokens`] array.
     /// @param feeFactor The fee factor to use for the trade.
-    /// @param encodedOrders Encoded signed EOA orders.
+    /// @param encodedTrades Encoded trades for signed EOA orders.
     /// @param encodedInteractions Encoded smart contract interactions.
     /// @param encodedOrderRefunds Encoded order refunds for clearing storage
     /// related to invalid orders.
     function settle(
         IERC20[] calldata tokens,
         uint256[] calldata clearingPrices,
         uint256 feeFactor,
-        bytes calldata encodedOrders,
+        bytes calldata encodedTrades,
         bytes calldata encodedInteractions,
         bytes calldata encodedOrderRefunds
     ) external view onlySolver {
         require(tokens.length == 0, "not yet implemented");
         require(clearingPrices.length == 0, "not yet implemented");
         require(feeFactor == 0, "not yet implemented");
-        require(encodedOrders.length == 0, "not yet implemented");
+        require(encodedTrades.length == 0, "not yet implemented");
         require(encodedInteractions.length == 0, "not yet implemented");
         require(encodedOrderRefunds.length == 0, "not yet implemented");
         revert("Final: not yet implemented");
     }
+
+    /// @dev Process all trades for EOA orders one at a time returning the
+    /// computed net in and out transfers for the trades.
+    ///
+    /// This method reverts if processing of any single trade fails. See
+    /// [`processOrder`] for more details.
+    /// @param tokens An array of ERC20 tokens to be traded in the settlement.
+    /// @param clearingPrices An array of token clearing prices.
+    /// @param encodedTrades Encoded trades for signed EOA orders.
+    /// @return inTransfers Array of transfers into the settlement contract.
+    /// @return outTransfers Array of transfers to pay out to EOAs.
+    function processTrades(
+        IERC20[] calldata tokens,
+        uint256[] calldata clearingPrices,
+        bytes calldata encodedTrades
+    )
+        internal
+        view
+        returns (
+            GPv2AllowanceManager.Transfer[] memory inTransfers,
+            GPv2AllowanceManager.Transfer[] memory outTransfers
+        )
+    {
+        uint256 tradeCount = encodedTrades.tradeCount();
+        inTransfers = new GPv2AllowanceManager.Transfer[](tradeCount);
+        outTransfers = new GPv2AllowanceManager.Transfer[](tradeCount);
+
+        GPv2Encoding.Trade memory trade;
+        for (uint256 i = 0; i < tradeCount; i++) {
+            encodedTrades.tradeAtIndex(i).decodeTrade(
```

Comment:
I see that this has made its way in here with or without the draft PR related to this change!

Reply:
Yes, that is on purpose. The draft PR was just to share my finding related to some unexpected Solidity compiler behaviour. I think its much nicer to read the code as is.